File: php/src/ApiInternal/DestinationAutoconfig.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\ApiInternal;

use Svix\Exception\ApiException;
use Svix\Models\DestinationIn;
use Svix\Models\DestinationOut;
use Svix\Request\SvixHttpClient;

class DestinationAutoconfig
{
    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
    }

    /**
     * Create or update the destination for an AutoConfig subscription.
     *
     * @throws ApiException
     */
    public function subscribe(
        string $appId,
        string $autoconfigId,
        DestinationIn $destinationIn,
    ): DestinationOut {
        $request = $this->client->newReq('PUT', "/api/v1/app/{$appId}/autoconfig/{$autoconfigId}/destination");
        $request->setBody(json_encode($destinationIn));
        $res = $this->client->send($request);

        return DestinationOut::fromJson($res);
    }
}

File: csharp/Svix/ApiInternal/DestinationAutoconfig.cs
// this file is @generated
#nullable enable
using Microsoft.Extensions.Logging;
using Svix.Models;

namespace Svix.ApiInternal
{
    public class DestinationAutoconfig(SvixClient client)
    {
        readonly SvixClient _client = client;

        /// <summary>
        /// Create or update the destination for an AutoConfig subscription.
        /// </summary>
        public async Task<DestinationOut> SubscribeAsync(
            string appId,
            string autoconfigId,
            DestinationIn destinationIn,
            CancellationToken cancellationToken = default
        )
        {
            destinationIn = destinationIn ?? throw new ArgumentNullException(nameof(destinationIn));
            try
            {
                var response = await _client.SvixHttpClient.SendRequestAsync<DestinationOut>(
                    method: HttpMethod.Put,
                    path: "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}/destination",
                    pathParams: new Dictionary<string, string>
                    {
                        { "app_id", appId },
                        { "autoconfig_id", autoconfigId },
                    },
                    content: destinationIn,
                    cancellationToken: cancellationToken
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(SubscribeAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Create or update the destination for an AutoConfig subscription.
        /// </summary>
        public DestinationOut Subscribe(
            string appId,
            string autoconfigId,
            DestinationIn destinationIn
        )
        {
            destinationIn = destinationIn ?? throw new ArgumentNullException(nameof(destinationIn));
            try
            {
                var response = _client.SvixHttpClient.SendRequest<DestinationOut>(
                    method: HttpMethod.Put,
                    path: "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}/destination",
                    pathParams: new Dictionary<string, string>
                    {
                        { "app_id", appId },
                        { "autoconfig_id", autoconfigId },
                    },
                    content: destinationIn
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(Subscribe)} failed");

                throw;
            }
        }
    }
}
